File: app/src/main/java/com/example/mobile_challenge/model/ResponseOptions.kt
package com.example.mobile_challenge.model

sealed class ResponseOptionsCurrency {
  data class SuccessResponse(val data: CurrencyResponse) : ResponseOptionsCurrency()
  data class ErrorResponse(val message: String) : ResponseOptionsCurrency()
}

sealed class ResponseOptionsQuote {
  data class SuccessResponse(val data: QuoteResponse) : ResponseOptionsQuote()
  data class ErrorResponse(val message: String) : ResponseOptionsQuote()
}

File: mobile-challenge/Model/ResponseOptions.swift
//
//  ResponseOptions.swift
//  mobile-challenge
//
//  Created by Kivia on 5/2/20.
//  Copyright © 2020 AP Club. All rights reserved.
//

import Foundation

enum ResponseOptionsCurrency {
  
  case SucessResponse(CurrencyResponse)
  case ErrorResponse(String)
  
}

enum ResponseOptionsQuote {
  
  case SucessResponse(QuoteResponse)
  case ErrorResponse(String)
  
}
